### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by HenryTabima**
My own version of enumerable methods with test cases as comments

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  def my_select
+    selected = []
+    my_each do |i|
+      selected << i if yield(i)
+    end
+    selected
+  end
+
+  def my_all?
+    my_inject(true) { |res, i| res && yield(i) }
```
Comment (zlayabekrija) [inline main.rb]: Please don't reuse the method that is returning a value.

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  end
+
+  def my_any?
+    my_inject(false) { |res, i| res || yield(i) }
```
Comment (zlayabekrija) [inline main.rb]: Same here, it's obvious that you understand the way everything works, which is great, yet methods shouldn't rely on top of another.

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  end
+
+  def my_none?
+    !my_any? { |i| yield(i) }
```
Comment (zlayabekrija) [inline main.rb]: What would happen if you delete the my_any? method, my_none wouldn't work either.

**Review by zlayabekrija — CHANGES_REQUESTED** [review]
This was a great showcase of your skills great work 👏, yet my major concern here is that your methods were built on top of another, please fix this stuff, I am sure it's going to be a piece of 🍰 for you, submit the new review request and there you go..
Happy coding 😄

**Review by zlayabekrija — APPROVED** [review]
Great work, great understanding. Thumbs up 👍 . Good luck with the next projects, and happy coding.

**Comment by HenryTabima** [comment]
Thanks! :+1:

**Pull request merged**